cranelift-codegen/src/regalloc/coloring.rs

```diff
@@ -226,7 +226,17 @@ impl<'a> Context<'a> {
                 lv.affinity.display(&self.reginfo),
                 self.cur.func.locations[lv.value].display(&self.reginfo)
             );
-            if let Affinity::Reg(rci) = lv.affinity {
+            if let Affinity::RegUnit(reg) = lv.affinity {
+                let rc = self.reginfo.toprc_containing_regunit(reg);
+                let loc = self.cur.func.locations[lv.value];
+                match loc {
+                    ValueLoc::Reg(reg) => regs.take(rc, reg, lv.is_local),
+                    ValueLoc::Unassigned => panic!("Live-in {} wasn't assigned", lv.value),
+                    ValueLoc::Stack(ss) => {
+                        panic!("Live-in {} is in {}, should be register", lv.value, ss)
+                    }
+                }
```

Comment:
I think I'd just pushed https://github.com/CraneStation/cranelift/pull/782/commits/ebdf736d9b602d311c6998502334e01d1f9c042d when you made this comment - done!

Reply:
Unless you mean getting the register class corresponding to an Affinity, which is a bit tricker. In some places it's correct to use the top `RegClass`, where in others `RegClassIndex` is used to get just the matching `RegClass`. I realized in the above change that it might actually be desirable to get the most precise matching class for a `RegUnit`, but either way a helper for `Affinity -> Option<RegClass>` seems good